src/hddsupport.c

```diff
@@ -93,16 +93,63 @@ static int hddCheckHDProKit(void)
 #define PFS_ZONE_SIZE 8192
 #define PFS_FRAGMENT 0x00000000
 
-static int CreateOPLPartition(const char *oplPart, const char *mountpoint)
+static int hddFindOPLPartition(void)
+{
+    static config_set_t *config;
+    char name[64];
+    int fd, ret = 0;
+
+    fileXioUmount(hddPrefix);
+
+    ret = fileXioMount(hddPrefix, "hdd0:__common", FIO_MT_RDWR);
+    if (ret == 0) {
+        fd = open("pfs0:conf_hdd.cfg", O_RDONLY);
+        if (fd >= 0) {
+            config = configAlloc(0, NULL, "pfs0:conf_hdd.cfg");
+            configRead(config);
+
+            configGetStrCopy(config, "hdd_partition", name, sizeof(name));
+            snprintf(oplPart, sizeof(oplPart), "hdd0:%s", name);
+
+            configFree(config);
+            close(fd);
+
+            if (!strcmp(name, "__common"))
+                return 0;
+
+            fileXioUmount(hddPrefix);
+            return 1;
+        }
+
+        fd = open("pfs0:conf_hdd.cfg", O_CREAT | O_TRUNC | O_WRONLY);
```

Comment:
hmmm, looks like you destroyed mount point when called `fileXioUmount` are you sure that `pfs0` contains something at this moment? I advise removing `fileXioUmount` cause you already have `fileXioUmount` at the end of the function.

Reply:
but when you umount it - pfs0 is destroyed, so you cannot open any files from pfs0

Reply:
Ahh I see you called umount before return - then it makes sense, didn't realize it.